rpc/src/server/mod.rs

```diff
@@ -124,27 +137,46 @@ pub fn spawn_server(bind_address: &SocketAddr, env: RpcServiceEnvironment) -> im
                     let env = env.clone();
                     let routes = routes.clone();
                     async move {
-                        if let Some((handler, params)) = routes.find(req.uri().path().to_string().trim_end_matches("/")) {
-                            match *req.method() {
-                                Method::OPTIONS => {
-                                    // lets globaly handle options
-                                    options()
-                                }
-                                _ => {
-                                    let params: Params = params.into_iter().map(|(param, value)| (param.to_string(), value.to_string())).collect();
-                                    let query: Query = req.uri().query().map(parse_query_string).unwrap_or_else(|| HashMap::new());
-
-                                    let handler = handler.clone();
-                                    let log = env.log.clone();
-                                    let fut = handler(req, params, query, env);
-                                    match Pin::from(fut).await {
-                                        Ok(response) => Ok(response),
-                                        Err(e) => {
-                                            error!(log, "Failed to execute RPC function - unhandled error"; "reason" => format!("{:?}", &e));
-                                            error_with_message(format!("{:?}", e))
+                        if let Some((method_and_handler, params)) = routes.find(req.uri().path().to_string().trim_end_matches("/")) {
+                            let MethodAndHandler {
+                                allowed_methods,
+                                handler,
+                            } = method_and_handler;
+
+                            let request_method = req.method();
+                            let log = env.log.clone();
+
+                            warn!(log, "REQUEST: {:?}", req.uri().path().to_string().trim_end_matches("/"));
+
+                            warn!(log, "REQUEST METHOD: {:?}", request_method);
+                            warn!(log, "Allowed METHODs: {:?}", allowed_methods);
+
+                            if allowed_methods.contains(&*request_method) {
+
+                                match *request_method {
+                                    Method::OPTIONS => {
```

Comment:
@adonagy 
this check `Method::OPTIONS` should be before `if allowed_methods.contains(&*request_method) {`
because it will never be triggered, because `Method::OPTIONS` is not configured for any route :)

Reply:
Oh rigth! :)